edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Agent.cs

```diff
@@ -145,6 +145,10 @@ public async Task ReconcileAsync(CancellationToken token)
                                 throw;
                             }
                         }
+                        else
+                        {
+                            status = Option.Some(DeploymentStatus.Success);
```

Comment:
Since `status` is now a `Some` in every path, why does it need to be `Option` at all? Maybe have it be `DeploymentStatus.Success` by default and only override it inside the `catch` block?